Makefile

```
@@ -58,25 +58,28 @@ generate: ## - Generate schema models
 	@printf "${CMD_COLOR_ON} Running go generate\n${CMD_COLOR_OFF}"
 	env PATH="${GOBIN}:${PATH}" go generate ./...
 
-.PHONY: check
-check: ## - Run all checks
+.PHONY: check-ci
+check-ci: ## - Run all checks of the ci without linting, the linter is run through github action to have comments in the pull-request.
 	@$(MAKE) generate
 	@$(MAKE) defaults
 	@$(MAKE) check-headers
-	@$(MAKE) check-go
 	@$(MAKE) notice
 	@$(MAKE) check-no-changes
 
+.PHONY: check
+check: ## - Run all checks
+	@$(MAKE) check-ci
+	@$(MAKE) check-go
+
 .PHONY: check-headers
 check-headers:  ## - Check copyright headers
 	@env GOBIN=${GOBIN} go install github.com/elastic/go-licenser@latest
 	@env PATH="${GOBIN}:${PATH}" go-licenser -license Elastic
 
 .PHONY: check-go
-check-go: ## - Run go fmt, go vet, go mod tidy
-	@go fmt ./...
-	@go vet ./...
-	@go mod tidy
+check-go: ## - Run golangci-lint
+	@curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s v1.44.2
```

I wonder if we can use a fixed version instead,for instance the last commit for https://github.com/golangci/golangci-lint/releases/tag/v1.44.2, https://github.com/golangci/golangci-lint/commit/d58dbde584c801091e74a00940e11ff18c6c68bd

```suggestion
	@curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/d58dbde584c801091e74a00940e11ff18c6c68bd/install.sh | sh -s v1.44.2
```

---

I was using their recommended way in their documentation to integrate with ci tool. I mean we could, but using the tag seems like solid approach.